oak_runtime/src/node/http/tests.rs

```diff
@@ -17,72 +17,127 @@
 use super::*;
 use maplit::hashmap;
 use oak_abi::{label::Label, proto::oak::application::ApplicationConfiguration};
-use std::{fs, thread};
+use std::{fs, option::Option, thread::JoinHandle};
+
+struct HttpServerTester {
+    runtime: RuntimeProxy,
+    server_node_thread_handle: Option<JoinHandle<()>>,
+    oak_node_simulator_thread_handle: Option<JoinHandle<()>>,
+    notify_sender: Option<tokio::sync::oneshot::Sender<()>>,
+}
+
+impl HttpServerTester {
+    fn new(port: u32, with_simulator_thread: bool) -> HttpServerTester {
+        let runtime = create_runtime();
+        let server_node = create_server_node(port);
+        let (init_receiver, invocation_receiver) = create_communication_channel(&runtime);
+
+        // Start the server node
+        let (notify_sender, notify_receiver) = tokio::sync::oneshot::channel::<()>();
+        let runtime_proxy = runtime.clone();
+        // TODO(#1186): Use tokio instead of spawning a thread.
+        let server_node_thread = std::thread::spawn(move || {
+            server_node.run(runtime_proxy, init_receiver, notify_receiver)
+        });
+
+        // Simulate an Oak node that responds with 200 (OK) to every request it receives
+        // TODO(#1186): Use tokio instead of spawning a thread.
+        let mut oak_node_simulator_thread = None;
```

Comment:
I think the mutable binding can be avoided, since `if / else` blocks are expressions in Rust (and not just statements).

Reply:
Good point. Thanks.